lib/SILOptimizer/Utils/ConstantFolding.cpp

```diff
@@ -1574,6 +1583,14 @@ void ConstantFolder::initializeWorklist(SILFunction &f) {
         continue;
       }
 
+      // Builtin.ifdef only replaced when not building the stdlib.
+      if (isApplyOfBuiltin(*inst, BuiltinValueKind::Ifdef)) {
+        if (!inst->getModule().isStdlibModule()) {
```

Comment:
This should depend on `FrontendOptions::ParseStdlib` and not on the module name.
Either you'll add `ParseStdlib` to SILOptions or just use `SILOptions::DisableReplacement` (which is used for the assert configuration).